### PR TITLE
Adding album art play/pause option for Dock icon (comment in #275)

### PR DESCRIPTION
**Pull request by paullj1**
- Adds album art option for Dock icon (As mentioned by @nobecutan)
- Adds new station information to Dock icon menu

### CODE REVIEW

**Comment by paullj1** [comment]
@nriley, I expect there will be some more issues with IB since I still haven't upgraded to Xcode 8... trying to wait for the final build to be released, but I might as well just do it at this point.

**Comment by nriley** [comment]
Hi, sorry I have not had a chance to review this yet. We are having a much busier time at the hospital this week than last and have not had much more than a few minutes to spare.  We'll get to this at the latest this weekend so we can have a release with your changes out by the time Sierra is released for good.

Thanks again for your contributions!

**Comment by paullj1** [comment]
No worries, no rush!  Just happy to help out!

File: Sources/HermesAppDelegate.m [hunk 1]
```diff
@@ -69,13 +69,9 @@ - (NSMenu *)applicationDockMenu:(NSApplication *)sender {
...
-    [menu addItemWithTitle:[NSString stringWithFormat:@"   %@", [song title]]
-                    action:nil
-             keyEquivalent:@""];
-    [menu addItemWithTitle:[NSString stringWithFormat:@"   %@", [song artist]]
-                    action:nil
-             keyEquivalent:@""];
+    [menu addItemWithTitle:nowPlaying.title action:nil keyEquivalent:@""];;
+    [menu addItemWithTitle:song.title action:nil keyEquivalent:@""];
+    [menu addItemWithTitle:song.artist action:nil keyEquivalent:@""];
     [menu addItem:[NSMenuItem separatorItem]];
```
Comment (nriley) [inline Sources/HermesAppDelegate.m]: Not sure why you're doing this?  The previous design imitates iTunes.

File: Sources/HermesAppDelegate.m [hunk 2]
```diff
@@ -558,30 +604,11 @@ - (IBAction) updateStatusItem:(id)sender {
...
-                operation:NSCompositeSourceOver
-                 fraction:1.0];
-      [icon unlockFocus];
-    }
+    icon = [self buildPlayPauseAlbumArtImage:size];
     
   } else {
     // Use color application image
-    icon = [[NSApp applicationIconImage] copy];
+    icon = [NSImage imageNamed:@"pandora"];
```
Comment (nriley) [inline Sources/HermesAppDelegate.m]: This is the same icon — not sure the point of hard-coding an icon name?

File: Sources/HermesAppDelegate.m [hunk 3]
```diff
@@ -69,13 +69,9 @@ - (NSMenu *)applicationDockMenu:(NSApplication *)sender {
...
   Song *song = [[playback playing] playingSong];
   if (song != nil) {
-    [menu addItemWithTitle:@"Now Playing" action:nil keyEquivalent:@""];;
-    [menu addItemWithTitle:[NSString stringWithFormat:@"   %@", [song title]]
-                    action:nil
-             keyEquivalent:@""];
-    [menu addItemWithTitle:[NSString stringWithFormat:@"   %@", [song artist]]
-                    action:nil
-             keyEquivalent:@""];
+    [menu addItemWithTitle:nowPlaying.title action:nil keyEquivalent:@""];;
```
Comment (nriley) [inline Sources/HermesAppDelegate.m]: Should mention I do like this change though — I had thought the window title would always be there to show the station name but of course that is not always the case.

File: Sources/Controllers/PreferencesController.h [hunk 4]
```diff
@@ -26,6 +26,7 @@
 #define OPEN_DRAWER                @"openDrawer"
 #define SORT_STATIONS              @"sortStations"
 #define GROWL_TYPE                 @"notificationType"
+#define DOCK_ICON_ALBUM_ART        @"dockIconAlbumArt"
```
Comment (nriley) [inline Sources/Controllers/PreferencesController.h]: Good job with the preferences changes — bindings are tricky!

File: Sources/HermesAppDelegate.m (hunk 2 above)
Comment (paullj1) [inline Sources/HermesAppDelegate.m]: Yeah, so if it changes to be the album art, then it's no longer the Hermes app icon.  Does that make sense?

File: Sources/HermesAppDelegate.m (hunk 1 above)
Comment (paullj1) [inline Sources/HermesAppDelegate.m]: So I guess I was just trying to imitate what was happening with the status bar icon.  Either way works for me, just let me know.

File: Sources/Controllers/PreferencesController.h (hunk 4 above)
Comment (paullj1) [inline Sources/Controllers/PreferencesController.h]: Thanks!

File: Sources/HermesAppDelegate.m (hunk 3 above)
Comment (paullj1) [inline Sources/HermesAppDelegate.m]: Yeah, I suppose I could add back in the indentation (like it's done with iTunes).  Again, just let me know!

File: Sources/HermesAppDelegate.m (hunk 1 above)
Comment (nriley) [inline Sources/HermesAppDelegate.m]: OK, I'll just put it back — no need to submit a new patch.

File: Sources/HermesAppDelegate.m (hunk 2 above)
Comment (nriley) [inline Sources/HermesAppDelegate.m]: D’ohh. Yeah, of course. Thanks.

**Pull request merged**